src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -186,6 +189,7 @@ public static BoundCall Synthesized(SyntaxNode syntax, BoundExpression? receiver
                     isDelegateCall: false,
                     expanded: false,
                     invokedAsExtensionMethod: false,
+                    receiverCloned: false,
```

Comment:
>receiverCloned: false, [](http://example.com/codeflow?start=20&length=22)

Is there a guarantee that this helper is never going to be used by initial binding? The default feels bug prone.

Reply:
Forgot to investigate this, will look. But also made the flag three-state and added an assert to verify we only use it if we set it - so such cases should be caught.